src/modules/my-positions/selectors/my-positions.js

```diff
@@ -10,5 +10,9 @@ export default function () {
 export const selectPositionsMarkets = createSelector(
   selectMarkets,
   selectAccountPositionsState,
-  (markets, positions) => (markets || []).filter(market => Object.keys(positions || {}).find(positionMarketID => market.id === positionMarketID))
+  (markets, positions) => {
+    if (!markets || markets.length === 0) return []
+    if (!positions || positions.length === 0) return []
+    return (markets || []).filter(market => positions[market.id] != null)
```

Comment:
I think this can just be:
```javascript
return markets.filter(market => positions[market.id] != null)
```
since you're checking for `!markets` already.